runtime/sema/type.go

```diff
@@ -693,7 +693,7 @@ func (t *OptionalType) QualifiedString() string {
 }
 
 func FormatOptionalTypeID[T ~string](elementTypeID T) T {
-	return T(fmt.Sprintf("%s?", elementTypeID))
+	return T(fmt.Sprintf("(%s)?", elementTypeID))
```

Comment:
Maybe we want to safe some space here and not have a 3 byte overhead for an optional type. 
Would a "prefix operator" variant be sufficient (non-ambiguous)?

Reply:
Are TypeIDs ever user-visible? If not then I think a prefix operator works but if they are then it's a bit hard to read.

Reply:
They do show up at times, mostly for developers, e.g. in the JSON encoding, so yeah, maybe we want to keep it readable